block/produce.go

```diff
@@ -228,3 +245,29 @@ func (m *Manager) createTMSignature(block *types.Block, proposerAddress []byte,
 	}
 	return vote.Signature, nil
 }
+
+// GetPreviousBlockHashes returns the hash of the last block and the commit for the last block
+// to be used as the previous block hash and commit for the next block
+func (m *Manager) GetPreviousBlockHashes(forHeight uint64) (lastHeaderHash [32]byte, lastCommit *types.Commit, err error) {
+	lastHeaderHash, lastCommit, err = loadPrevBlock(m.Store, forHeight-1)
+	if err != nil {
+		if !m.State.IsGenesis() { // allow prevBlock not to be found only on genesis
+			return [32]byte{}, nil, fmt.Errorf("load prev block: %w: %w", err, ErrNonRecoverable)
+		}
+		lastHeaderHash = [32]byte{}
+		lastCommit = &types.Commit{}
+	}
+	return lastHeaderHash, lastCommit, nil
+}
```

Comment:
can you explain the +-1 stuff going on here a bit more clearly

So there are three blocks

A,B,C
A is last
C is next?

or A,B
A is last
B is next?

Reply:
`GetPreviousBlockHashes` for height `h` returns the hashes of block `h-1`

Reply:
so the second thing I said
can you clarify in the docstring?